### PR TITLE
[Merged by Bors] - Return HTTP 404 rather than 405

### PR DESCRIPTION
**Pull request by santiagomed**
## Issue Addressed

Issue #3112

## Proposed Changes

Add `Filter::recover` to the GET chain to handle rejections specifically as 404 NOT FOUND

## Additional Info

Making a request to `http://localhost:5052/not_real` now returns the following:

```
{
    "code": 404,
    "message": "NOT_FOUND",
    "stacktraces": []
}
```

### CODE REVIEW

**Review by paulhauner — CHANGES_REQUESTED** [review]
Great, thanks for this contribution @santiagomed! It really irks me how our HTTP server gives weird error codes 😥

Your fix is simple and works well. I was wondering if you could do the same for POST requests too, please? I tried this locally and it should be as simple as adding the same line here: https://github.com/sigp/lighthouse/blob/260f78aebc46feae4bdf6414e42369b83a7805db/beacon_node/http_api/src/lib.rs#L3380

**Comment by paulhauner** [comment]
I've marked this PR as `backwards-incompat` because I think it deserves a mention in our release notes.

**Comment by santiagomed** [comment]
@paulhauner made the addition! Let me know if anything else is needed.

File: beacon_node/http_api/src/lib.rs [hunk 1]
```diff
@@ -3377,6 +3378,7 @@ pub fn serve<T: BeaconChainTypes>(
                 .or(post_lighthouse_database_reconstruct.boxed())
                 .or(post_lighthouse_database_historical_blocks.boxed())
                 .or(post_lighthouse_block_rewards.boxed()),
```
Comment (paulhauner) [inline beacon_node/http_api/src/lib.rs]:
```suggestion
                .or(post_lighthouse_block_rewards.boxed())
```

**Review by paulhauner — CHANGES_REQUESTED** [review]
Thanks! It looks like there's been a conflict sorry, I imagine it'll be simple to resolve.

I've also made a suggest to what I *think* will be a compile error.

**Comment by paulhauner** [comment]
I just tried my hand at resolving the conflict and compile error with the Github UI, let's see how that goes!

File: beacon_node/http_api/src/lib.rs (hunk 1 above)
Comment (paulhauner) [reply beacon_node/http_api/src/lib.rs]: Please ignore this, I fixed this when resolving the conflict.

**Review by paulhauner — APPROVED** [review]
It passed CI 🎉 Thanks again @santiagomed. This is good to merge once v3.4.0 is released and proven stable. Until then, it's `blocked`.

**Comment by paulhauner** [comment]
### Breaking changes note

Previously, the Beacon Node HTTP would return `405` (method not allowed) for an unknown route (e.g. `http://localhost:5052/not_a_real_route`). Lighthouse will now return a `404` (not found) error.

**Pull request closed without merging**